packages/core/types/stitches.d.ts

```diff
@@ -204,7 +204,7 @@ type ThemeTokens<Values, Prefix> = {
 	[Scale in keyof Values]: {
 		[Token in keyof Values[Scale]]: ThemeUtil.Token<
 			Extract<Token, number | string>,
-			Values[Scale][Token],
+			Values[Scale][Token] extends string | number ? Values[Scale][Token] : string,
```

Comment:
NOTE: This change is just inherited from #893, and not directly related to this PR.